analyzer/src/main/kotlin/managers/SBT.kt

```diff
@@ -80,83 +80,54 @@ class SBT(config: AnalyzerConfiguration) : PackageManager(config) {
     override fun prepareResolution(definitionFiles: List<File>): List<File> {
         if (definitionFiles.isEmpty()) return emptyList()
 
+        val workingDir = if (definitionFiles.count() > 1) {
+            // Some SBT projects do not have a build file in their root, but they still require "sbt" to be run from the
+            // project's root directory. In order to determine the root directory, use the common prefix of all
+            // definition file paths.
+            val projectRoot = definitionFiles.map {
+                it.absolutePath
+            }.reduce { prefix, path ->
+                prefix.commonPrefixWith(path)
+            }
+
+            File(projectRoot).also {
+                log.info { "Determined '$it' as the ${toString()} project root directory." }
+            }
+        } else {
+            definitionFiles.first().parentFile
+        }
+
         // Note that "sbt sbtVersion" behaves differently when executed inside or outside an SBT project, see
         // https://stackoverflow.com/a/20337575/1127485.
-        var workingDir = definitionFiles.first().parentFile
-
-        // We need at least sbt version 0.13.0 to be able to use "makePom" instead of the deprecated hyphenated form
-        // "make-pom" and to support declaring Maven-style repositories, see
-        // http://www.scala-sbt.org/0.13/docs/Publishing.html#Modifying+the+generated+POM.
         checkCommandVersion(
```

Comment:
I'd try to check the `project/build.properties` first, because it will probably be there in 99% of the cases and will be much faster than starting sbt just to get its version. This is an officially recommended approach: https://www.scala-sbt.org/1.0/docs/Basic-Def.html#Specifying+the+sbt+version, and it's also how the sbt itself checks its current version: https://github.com/sbt/sbt/blob/1.x/main/src/main/scala/sbt/Main.scala#L802-L811. If the `project/build.properties` file is missing, then we can fallback to this method. This can be done as a separate change though, so it's not blocking the current review.

Reply:
Agreed, I'll do that in a new PR. Thanks again!

Reply:
Actually, I decided to just add this performance optimization to this PR, too, because it also speeds up my testing :-)